cdc/sink/producer/kafka/kafka.go

```diff
@@ -197,26 +213,27 @@ func (k *kafkaSaramaProducer) GetPartitionNum() int32 {
 }
 
 // stop closes the closeCh to signal other routines to exit
+// It SHOULD NOT be called under `clientLock`.
 func (k *kafkaSaramaProducer) stop() {
-	k.clientLock.Lock()
-	defer k.clientLock.Unlock()
-	select {
-	case <-k.closeCh:
+	if atomic.SwapInt32(&k.closing, kafkaProducerClosing) == kafkaProducerClosing {
```

Comment:
The logic here has a double meaning, will it be easier to understand if you separate them?

Reply:
This logic cannot be divided into two steps, I am wrong.